### ISSUE TITLE
[BUG] Rancher API Audit Log fails to capture API calls at Audit Level 3

### ISSUE THREAD

**Issue by samjustus** [rancher/rancher#38323]
**Rancher Server Setup**
- Rancher version: 2.5.12, 2.6.5

**Describe the bug**
The Rancher audit logs fail to capture important audit events when an API call is made from the UI to apply a resource change, e.g. when a StatefulSet or Deployment is created, re-deployed, and scaled up or down from the Rancher UI.

In contrast, audit events in relation to these k8s operations are recorded when the audit log level is set to 2 in the helm chart options for Rancher.

**To Reproduce**
Enable Rancher API audit log level 3 with the helm chart install
Perform Kubernetes operations from the Rancher UI, e.g. create a new Deployment or Pod from YAML, scale up/down a Deployment or select the 'Redeploy' option

**Result**
A number of API events involving methods POST/PUT do not get recorded. This is evident when the log level is set to 3.

**Expected Result**
All Kubernetes API events should be recorded when the log level is set to 3.

**Comment by KevinJoiner** [comment]
@niusmallnan I didn't see your PR, but this is the same issue. @samjustus This should get backported to 2.5.

**Comment by samjustus** [comment]
SURE 4837

**Comment by anupama2501** [comment]
QA to validate:

- Enabling audit logs as follows for RKE1 clusters:
```
services:
  kube-api:
    audit_log:
      enabled: true
```
- Fresh checks - Verify the audit logs as a standard user and audit log level set to 3,2,1
  - Creating and upgrading an app
  - Creating a deployment with 4 replicas - Scale down and scale up the replicas verify audit logs
  - Deleting a deployment
  - Re deploy a deployment
  - Create a stateful set - scale up, redeploy, delete
  - Verify no error logs seen in rancher logs.
  - Add users in a cluster - verify the logs are recorded
  - Enable auth provider and change the auth provider
- Upgrade checks - Reproduce the issue on v2.6.7
  - Before upgrade, create a few deployments, stateful sets and ingress - run pre upgrade checks jenkins job
  - Post upgrade - run post upgrade jenkins job which creates a few deployments, ingress, app, redeploys the deployment
  - Delete the deployment

**Comment by anupama2501** [comment]
Reproduced on a docker install v2.6.7
1. Install rancher docker run -d --restart=unless-stopped  -p 80:80 -p 443:443 -v /var/log/rancher/auditlog:/var/log/auditlog -e AUDIT_LEVEL=3 --privileged  rancher/rancher:v2.6.7
2. Created a downstream rke1 and rke2 node driver linode clusters
3. Created a deployment, a stateful set, redeployed the deployment, scaled up the deployment - None of the events were recorded.

**Comment by anupama2501** [comment]
Re opening the issue as for the following test cases audit logs are not captured when Audit log is set to level 3
On rancher v2.6-head 24f8f91, HA 

**Steps to reproduce:**
1. Install rancher server via helm and while installing add the env variable `--set auditLog.level=3`
2. As an admin, create 1 rke1 node driver downstream cluster and an rke2 node driver downstream cluster with 1 etcd, 1 cp and 1 worker pools in each of the clusters.
3. From the rke1 and rke2 clusters, created a deployment and a daemon set with the container image `ranchertest/mytestcontainer`.   

Notice an error in rancher pod logs for all the below cases:
```
2022/09/08 10:35:06 [WARNING] Failed to write audit log: failed to compact audit log: invalid character '{' after object key:value pair
```
**Test cases on RKE1 node driver, k8s : v1.24.4** 

1. Edit config - Edit the config and change the deployment replica count to 3 no audit logs were recorded
2. Scale down - Scale down the replicas after the above test case- no audit logs were recorded
3. Delete daemon set - Delete the daemon set created above and notice no audit logs recorded, for deployment deletion audit logs were recorded
4. Cloning - Clone the deployment and notice the audit log for  https://<URL>/k8s/clusters/cluster-id/v1/apps.deployments/ns3/wk2rke1 is not recorded.

**Test cases on RKE2 node driver, k8s : v1.24.4+rke2r1** 
1. Create deployment - for create deployment the POST request URI https://<URL>/k8s/clusters/cluster-id/v1/apps.deployments/ns2/wk2rke2 is not recorded
2. Re deploy deployment - No audit logs were recorded when we redeploy the deployment
3. Scale up deployment - When we scale up the replicas no audit logs were recorded
3. Delete deployment -  No audit logs were recorded when the deployment is deleted.
4. Clone deployment - for cloning similar to create deployment, the POST request URI https://<URL>/k8s/clusters/cluster-id/v1/apps.deployments/ns2/wk2rke2 is not recorded
5. Edit config daemon set - No audit logs were recorded when the daemon set config is edited and the replica count is set to a higher value.

**Comment by anupama2501** [comment]
Validated test cases so far on v2.6-head 24f8f91 audit log level set to 3,2
On rancher HA helm install and docker install
Setup:
1. Create 2 downstream node driver clusters RKE1, RKE2, standard users - user1, user2
2. Add user1 as cluster owner in both the clusters and add user2 as cluster member in both the clusters.

Ran the following validations:
1. Create deployment - for create deployment the POST request URI https://<URL>/k8s/clusters/cluster-id/v1/apps.deployments/ns2/wk2rke2 - PASS
2. Re deploy deployment - PASS
3. Scale up deployment - PASS
4. Delete deployment -  PASS
5. Clone deployment - PASS
6. Edit config daemon set - PASS
7. Scale down - PASS
8. Create daemon set - PASS
9. Edit config for daemon set - no audit logs - PASS
10. Delete daemon set - PASS


Installed rancher cli version 2.6.9-rc1, and ran the following cases as a cluster member and cluster owner and verified the audit logs recorded for the following cases:

1. Create deployment -  PASS
2. Re deploy deployment - PASS
3. Delete deployment -  PASS
4. Edit config daemon set - PASS

**Comment by KevinJoiner** [comment]
Merged Before the 2.7 Branch

**Comment by KevinJoiner** [comment]
## Synopsis of issue
When API Audit logging for Rancher Manager **_not from kube-apiserver_** was set to level 3 verbosity, which includes response data, messages would be dropped if the response data was compressed.
## Resolved (or not):
This issue was resolved, and now audit log messages are properly decompressed.
## Status: 
To be resolved in v2.6.9 .
## Versions: 
<= v2.6.8
